Library/Homebrew/linkage_checker.rb

```diff
@@ -196,7 +196,14 @@ def check_dylibs(rebuild_cache:)
         rescue Errno::ENOENT
           next if harmless_broken_link?(dylib)
 
-          if (dep = dylib_to_dep(dylib))
+          basename = File.basename(dylib)
+          if dylib_found_via_dlopen(basename)
+            # `dyld` will fall back to its default search path when it cannot
+            # resolve an install name. Calling `dlopen` on the basename simulates
+            # this search procedure. If this succeeds, the formula is unlikely
+            # to be broken.
+            @system_dylibs << basename
+          elsif (dep = dylib_to_dep(dylib))
             @broken_deps[dep] |= [dylib]
           elsif MacOS.version >= :big_sur && dylib_found_via_dlopen(dylib)
```

Comment:
This part is still necessary, since System Frameworks typically cannot be found by their basename.